### PR TITLE
#1102 - Embedding-based product similarity search

### PR DESCRIPTION
**Pull request by duylv27**

### CODE REVIEW

File: product/src/main/java/com/yas/product/service/ProductRecommendationService.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+import org.springframework.transaction.annotation.Transactional;
+
+/**
+ * Service class responsible for managing product recommendations.
+ */
+
+@Service
+@Transactional
+@Slf4j
+public class ProductRecommendationService {
```
Comment (duylv27) [inline product/src/main/java/com/yas/product/service/ProductRecommendationService.java:33]: This class provides product data to the recommendation service, so the name `ProductRecommendationService` does not seem appropriate.

File: product/src/main/java/com/yas/product/service/ProductRecommendationService.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+import java.util.Optional;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+/**
+ * Service class responsible for managing product recommendations.
+ */
+
```
Comment (duylv27) [inline product/src/main/java/com/yas/product/service/ProductRecommendationService.java:29]: Remove this redundant line

File: product/src/main/java/com/yas/product/service/ProductRecommendationService.java [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        List<ProductAttributeValueGetVm> productAttributes = product.getAttributeValues()
+                .stream()
+                .map(ProductAttributeValueGetVm::fromModel)
+                .toList();
+
+        if (Boolean.TRUE.equals(product.isHasOptions())) {
+            List<Product> productVariations = product.getProducts()
+                    .stream()
+                    .toList();
+            variations = productVariations.stream()
```
Comment (duylv27) [inline product/src/main/java/com/yas/product/service/ProductRecommendationService.java:91]: Product variations are considered separate products, but the recommendation service expects to receive a product corresponding to a `long productId`. Is there any specific reason to retrieve variations along with the requested product? Let's discuss.

**Review by khanhtrand — APPROVED** [review]

File: product/src/main/java/com/yas/product/service/ProductRecommendationService.java (hunk 3 above)
Comment (duylv27) [reply product/src/main/java/com/yas/product/service/ProductRecommendationService.java]: Temporarily hold this discussion. Product variations will be handled in [#1149](https://github.com/nashtech-garage/yas/issues/1149), then this API/service will no longer be needed.

**Review by PhuocNguyenBa — APPROVED** [review]

**Pull request merged**